### PR TITLE
Use urlquote for fast (un)quoting

### PR DESCRIPTION
**Pull request by pacman82**
# Description:

Use urlquote for fast quoting/unquoting. Two side effects are known:

* Seems to also fix a minor edge case there if a binary column is used for partitioning, it might raise a `UnicodeDecodeError`. See changes in `test_metapartition.py`
* `~` is now not going to be percent encoded on all Python versions. Previously it has been encoded in Python < 3.7.

- [x] Tests added / passed
- [x] Passes `./format_code.sh`
- [x] Changelog entry

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -9,6 +9,7 @@ Version 3.2.1 (2019-XX-XX)
 - Streamline behavior of `store_dataset_from_ddf` when passing empty ddf.
 - Fix an issue where a segmentation fault may be raised when comparing MetaPartition instances
 - Expose a ``date_as_object`` flag in ``kartothek.core.index.as_flat_series``
+- Use urlquote for fastor quoting/unquoting
```
Comment (fjetter) [inline CHANGES.rst]: Typo: faster

Comment (pacman82) [reply CHANGES.rst]: done

**Comment by fjetter** [comment]
Is urlquote available as a Conda forge package?

**Comment by xhochy** [comment]
@fjetter @pacman82 Is there is a link to the conda-forge/staged-recipes PR for the urlquote conda package?

**Comment by pacman82** [comment]
@xhochy Just working on one: https://github.com/pacman82/staged-recipes. Had trouble testing in on my private windows machine. Also my first conda package, so any help and feedback is welcome.

**Comment by xhochy** [comment]
Best thing would be to make a WIP/Draft-PR to conda-forge and cc me there. Then we can iterate there with CI and commenting support.

**Comment by pacman82** [comment]
PR to conda-forge: https://github.com/conda-forge/staged-recipes/pull/9194

**Comment by pacman82** [comment]
Now there is a feedstock. Thanks for the help, @xhochy!

**Comment by pacman82** [comment]
ping @fjetter anything missing for this to be merged?

File: CHANGES.rst [hunk 2]
```diff
@@ -9,6 +9,7 @@ Version 3.2.1 (2019-XX-XX)
 - Streamline behavior of `store_dataset_from_ddf` when passing empty ddf.
 - Fix an issue where a segmentation fault may be raised when comparing MetaPartition instances
 - Expose a ``date_as_object`` flag in ``kartothek.core.index.as_flat_series``
+- Use urlquote for faster quoting/unquoting
```
Comment (crepererum) [inline CHANGES.rst]: please rebase and create a new version (3.4.0) for this change

**Comment by crepererum** [comment]
I've tried out the "resolve conflict" feature in the GitHub UI. It's kinda messy and creates ugly commits. Waiting for the tests once more and will merge afterwards.

**Pull request merged**